controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -545,3 +562,46 @@ func firstNodeNotMatchingName(name string, nodes []corev1.Node) *corev1.Node {
 	}
 	return nil
 }
+
+// modifyImageTag takes an image string, and returns an updated tagged image
+func modifyImageTag(image, tagName string) (string, error) {
+	namedRef, err := reference.ParseNormalizedNamed(image)
```

Comment:
It doesn't look like `err` is checked after this call, adding a check would resolve the last linter error.